org.eclipse.jdt.core/search/org/eclipse/jdt/internal/core/search/matching/MatchLocator.java

```diff
@@ -1012,27 +1016,25 @@ public MethodBinding getMethodBinding(MethodPattern methodPattern) {
     } else if (methodPattern.focus instanceof BinaryMethod &&
     		methodPattern.declaringType instanceof BinaryType &&
     		this.unitScopeTypeBinding instanceof ProblemReferenceBinding) {//Get binding from unit scope for non-visible member of binary type
-    	char[] typeName = PatternLocator.qualifiedPattern(methodPattern.declaringSimpleName, methodPattern.declaringQualification);
-    	if (typeName != null) {
```

Comment:
Removing this part might cause different behavior, but to me its dead code. `typeName` is not read by this block, just set below (maybe code was deleted at some point, or this was copy-paste).

Reply:
agree lets remove, we can visit if something goes wrong in ibuilds.